traffic_ops/traffic_ops_golang/server/detail.go

```diff
@@ -91,7 +91,11 @@ func GetDetailParamHandler(w http.ResponseWriter, r *http.Request) {
 				routerPortName = interfaces[0].RouterPortName
 			}
 			v11server := tc.ServerDetailV11{}
-			v11server.ServerDetail = server.ServerDetail
+			v11server.ServerDetail, err = dbhelpers.GetServerDetailFromV4(server, inf.Tx.Tx)
+			if err != nil {
+				api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, nil, errors.New("failed to GetServerDetailFromV4: "+err.Error()))
```

Comment:
This error is being passed up as an internal error and nothing is sent to the client - this should probably be returning a `500 Internal Server Error` response not `400 Bad Request`.

Also errors constructed from errors should wrap their source error using the `%w` formatting parameter of `fmt.Errorf`.

Reply:
@ocket8888: I don't understand, why I need to use `fmt.Errorf` instead of `errors.New`. In my understanding fmt gives me a string format and errors gives me error type.
I agree on the status code change.

Reply:
`fmt.Errorf` returns an `error` type, not a `string`. The reason is because when you make a new error by calling `err.Error()`, you're discarding the error and only carrying its string representation forward.

In this playground I constructed the "same" error 5 different ways: https://go.dev/play/p/oqjjeO4DcjA
When you run it, you can see that the identity of the underlying error is only preserved in the first case - using `fmt.Errorf` with `%w`.

Reply:
I see.. thanks for the explanation.